### DISCUSSION TITLE
[Detector Support]: Does Frigate "Fail To Detect" just as much as it "False Detects"?

### DISCUSSION THREAD

**Discussion by sharpcall** [blakeblackshear/frigate#21397] [Detector Support]
### Describe the problem you are having

There are many instances that Frigate capture inanimate object movements as live ones falsely. This is annoying but not a security risk mostly. However, are there just as many Fail to Detect instances too where Frigate does not detect a person for example when come into a zone?

### Version

0.17 beta

### Frigate config file

```yaml
NA
```

### docker-compose file or Docker CLI command

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Install method

Docker Compose

### Object Detector

Coral

### Screenshots of the Frigate UI's System metrics pages

NA

**Comment by inkeep-ai-assistant** [comment]
The available documentation doesn't directly compare false detection rates versus missed detection rates (failure to detect) for Frigate. However, I can provide context about how Frigate's detection system works and factors that influence accuracy.

**Detection Architecture:**

Frigate uses a two-stage process - motion detection triggers object detection[(1)](https://docs.frigate.video/frigate/hardware). The CPU monitors for pixel changes (motion), then sends those frames to your detector (like your Coral) for object identification[(1)](https://docs.frigate.video/frigate/hardware). This means both false positives and missed detections can occur at different stages.

**Factors Affecting Accuracy:**

The accuracy depends significantly on your hardware and model choice. For example, with a Coral EdgeTPU detector, you're using TensorFlow Lite models[(2)](https://docs.frigate.video/configuration/object_detectors). The default model provided at `/edgetpu_model.tflite` is optimized for the EdgeTPU[(2)](https://docs.frigate.video/configuration/object_detectors).

**Improving Detection Accuracy:**

To reduce both false positives and missed detections, consider:

1. **Zones** - Configure zones to restrict where detections matter, reducing false positives from irrelevant areas[(3)](https://docs.frigate.video/configuration/review)
2. **Review Configuration** - Set up alerts and detections with specific labels and required zones[(3)](https://docs.frigate.video/configuration/review)
3. **Motion Sensitivity** - Adjust motion detection thresholds in your configuration[(4)](https://docs.frigate.video/configuration/reference)
4. **Model Selection** - Different models have different accuracy characteristics[(2)](https://docs.frigate.video/configuration/object_detectors)

**Configuration Example for Filtering:**

```yaml
review:
  alerts:
    labels:
      - person
    required_zones:
      - front_door
```
[(3)](https://docs.frigate.video/configuration/review)

This configuration would only create alerts for person detections in the front_door zone, helping reduce false positives while focusing on what matters[(3)](https://docs.frigate.video/configuration/review).

Without specific metrics from your setup, I cannot quantify whether you're experiencing equal rates of false positives versus missed detections. However, both types of errors are influenced by your hardware capabilities, model selection, motion sensitivity settings, and zone configurations.

**

📚 **Sources:**
1. [Recommended hardware](https://docs.frigate.video/frigate/hardware)
2. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
3. [Review](https://docs.frigate.video/configuration/review)
4. [Full Reference Config](https://docs.frigate.video/configuration/reference)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by NickM-27** [comment]
Your question is way too broad. Firstly, it is not a matter of what Frigate "detects" or "fails to detect" as Frigate just runs the model you configure. There are a wide range of models available from free models trained on the COCO dataset to purpose built paid models that you can tune for your cameras, Frigaet+

Whether you have false positives (detections that are wrong) or false negatives (detections that missed) depends on the model you use, as well as your configuration. You have full control to tune the filters to ensure objects are detected more readily but at risk of more false positives, or to reduce false positives at the risk of false negatives. This balance changes depending on which model you run.

**Reply by sharpcall** [reply]
So the only tool available to have less False Negatives is motion sensitivity and the the type of model used. Period. Did I get that correct?
And lowering the sensitivity to lower False Negatives will potentially increase the False Positives.

What do you mean by "purpose built paid models that you can tune for your cameras, Frigate+". Why is this specific to the type of camera or even the environment? I mean a person is a person no matter which backyard they are in and the cameras overall should show a similar close enough representation of a person regardless of camera brand and aperture, etc.

Thanks for the explanation in advance.

**Reply by NickM-27** [reply]
> So the only tool available to have less False Negatives is motion sensitivity and the the type of model used. Period. Did I get that correct?

No, you also have filters. You can increase the sensitivity of the filters that are applied by default https://docs.frigate.video/configuration/object_filters

> What do you mean by "purpose built paid models that you can tune for your cameras, Frigate+". Why is this specific to the type of camera or even the environment? I mean a person is a person no matter which backyard they are in and the cameras overall should show a similar close enough representation of a person regardless of camera brand and aperture, etc.

The models that are used by Frigate by default are the freely available COCO models. They're trained on images that look nothing like the perspective of a security camera. For example some images show a "person" but it is just a picture of their hand.

Naturally, a model performs much better when it is trained on data similar to what it's being used for.  

Also you're wrong about a person being a person in the backyard. There are so many variables. For example, to name a few:

- a camera mounted above a door will have a very different perspective of a person compared to a camera mounted at the second story roof line
- some cameras are infra-red at night while others are color at night
- fine-tuning a model is not just teaching it what an object looks like but also what in the scene is not an object, which allows reducing false positives

If you want to read more https://frigate.video/plus/

**Reply by sharpcall** [reply]
Thanks.

Would it be possible to feed an empty backyard image to Frigate and then tell it that any difference from this image that is a certain size or bigger needs detection, alert, and recording? This might be an easier task to do for computers?

**Reply by NickM-27** [reply]
That would also be very complicated. For example, what about when a cloud casts a shadow or the sun changes and there is a shadow from part of the house or a tree? This is indeed a change but not a change that you care about. 

Not to mention weather changes like snow, rain, etc. These are all vastly different, so you can't really detect something changed specifically that you care about by just comparing pixels (which is motion detection). Hence, why object detection is useful.
